framework/src/variables/MooseVariableData.C

```diff
@@ -739,15 +741,56 @@ MooseVariableData<OutputType>::computeValues()
     }
   }
 
+  if (_has_scaled_grad)
+  {
+    scaleGrad(_grad_u);
+    if (_need_grad_old)
+      scaleGrad(_grad_u_old);
+    if (_need_grad_older)
+      scaleGrad(_grad_u_older);
+    if (_need_grad_previous_nl)
+      scaleGrad(_grad_u_previous_nl);
+    if (_need_grad_dot || _need_grad_dotdot)
+      mooseError("No support yet for gradient scaling with grad_dot");
```

Comment:
I'm not sure if gradient scaling screws up grad_dot, I just have this in here to be safe.